src/session.rs

```diff
@@ -127,6 +127,18 @@ impl Session {
         }
     }
 
+    /// Get the remote static key that was possibly encrypted in the first payload
+    ///
+    /// # Caveat
+    ///
+    /// This currently does not work *after* transitioning into the transport state.
+    pub fn get_remote_static(&self) -> Option<&[u8]> {
+        match *self {
+            Session::Handshake(ref state) => Some(state.get_remote_static().unwrap()),
```

Comment:
unwraps/panics are bad news, why do we need that? HandshakeState::get_remote_static() should return an Option directly, since there are Noise configurations where there is no remote static key.

Reply:
yeah you are right about the unwrap... here's the type incompatibility i found:
    = note: expected type `std::option::Option<&[u8]>`
               found type `std::option::Option<&[u8; 56]>`